clang/include/clang/CIR/Dialect/Builder/CIRBaseBuilder.h

```diff
@@ -408,6 +408,20 @@ class CIRBaseBuilderTy : public mlir::OpBuilder {
                            callee.getFunctionType().getReturnType(), operands);
   }
 
+  struct BinOpOverflowResults {
+    mlir::Value result;
+    mlir::Value overflow;
+  };
+
+  BinOpOverflowResults createBinOpOverflowOp(mlir::Location loc,
```

Comment:
I think it would be better to eliminate this function and just call `cir::BinOpOverflowOp::create` directly where it is currently used. The added `BinOpOverflowResults` structure feels less natural than just calling `getResult` and `getOverflow` on the returned operation.